src/zenml/integrations/gcp/orchestrators/vertex_orchestrator.py

```diff
@@ -621,6 +656,143 @@ def dynamic_pipeline() -> None:
             schedule=snapshot.schedule,
         )
 
+    def submit_dynamic_pipeline(
+        self,
+        snapshot: "PipelineSnapshotResponse",
+        stack: "Stack",
+        environment: Dict[str, str],
+        placeholder_run: Optional["PipelineRunResponse"] = None,
+    ) -> Optional[SubmissionResult]:
+        """Submits a dynamic pipeline to the orchestrator.
+
+        Args:
+            snapshot: The pipeline snapshot to submit.
+            stack: The stack the pipeline will run on.
+            environment: Environment variables to set in the orchestration
+                environment.
+            placeholder_run: An optional placeholder run.
+
+        Raises:
+            RuntimeError: If the snapshot contains a schedule.
+
+        Returns:
+            Optional submission result.
+        """
+        if snapshot.schedule:
+            raise RuntimeError(
+                "Scheduling dynamic pipelines is not supported for the "
+                "Vertex orchestrator yet."
+            )
+
+        settings = cast(
+            VertexOrchestratorSettings, self.get_settings(snapshot)
+        )
+
+        command = (
+            DynamicPipelineEntrypointConfiguration.get_entrypoint_command()
+        )
+        args = DynamicPipelineEntrypointConfiguration.get_entrypoint_arguments(
+            snapshot_id=snapshot.id,
+            run_id=placeholder_run.id if placeholder_run else None,
+        )
+
+        image = self.get_image(snapshot=snapshot)
+        labels = settings.labels.copy()
+        labels["source"] = f"zenml-{__version__.replace('.', '_')}"
+
+        job_request = build_job_request(
+            display_name=get_orchestrator_run_name(
+                pipeline_name=snapshot.pipeline_configuration.name
+            ),
+            image=image,
+            entrypoint_command=command + args,
+            custom_job_settings=settings.custom_job_parameters
+            or VertexCustomJobParameters(),
+            resource_settings=snapshot.pipeline_configuration.resource_settings,
+            environment=environment,
+            labels=labels,
+            encryption_spec_key_name=self.config.encryption_spec_key_name,
+            service_account=self.config.workload_service_account,
+            network=self.config.network,
+        )
+
+        _, project_id = self._get_authentication()
+        client = self.get_job_service_client()
```

Comment:
this ends up calling `_get_authentication` twice, which has the unfortunate effect that it uses the service connector twice to generate credentials. If you need the project ID, you should perhaps store it as a local variable and set it when you call `get_job_service_client`